### PR TITLE
Improve Build and Release Tasks

### PR DESCRIPTION
**Pull request by MadhukaHarith92**
## Purpose
Related to ballerina-platform/ballerina-standard-library#1618
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -45,19 +45,9 @@ jobs:
...
-                  "name": "module-ballerina-random-v'"$VERSION"'"
-                  }'
-            - name: Post release PR
+            - name: GitHub Release and Release Sync PR
               env:
                   GITHUB_TOKEN: ${{ secrets.BALLERINA_BOT_TOKEN }}
               run: |
-                  curl -fsSL https://github.com/github/hub/raw/master/script/get | bash -s 2.14.1
-                  bin/hub pull-request --base main -m "[Automated] Sync main after "$VERSION" release"
+                gh release create v$VERSION --title "module-ballerina-email-v$VERSION"
```
Comment (ThisaruGuruge) [inline .github/workflows/publish-release.yml:52]:
```suggestion
                gh release create v$VERSION --title "module-ballerina-random-v$VERSION"
```

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -1,12 +1,6 @@
 name: PR build
 
-on:
-  pull_request:
-  push:
-    branches-ignore:
-      - main
-      - "automated/dependency_version_update"
-      - "automated/dependency_version_update_tmp"
+on: [pull_request]
```
Comment (ThisaruGuruge) [inline .github/workflows/pull-request.yml:3]:
```suggestion
on: pull_request
```

**Review by ThisaruGuruge — COMMENTED** [review]
Gradle Wrapper jar is not pushed. Please check

File: .github/workflows/publish-release.yml [hunk 3]
```diff
@@ -45,19 +45,9 @@ jobs:
...
-                  }'
-            - name: Post release PR
+            - name: GitHub Release and Release Sync PR
               env:
                   GITHUB_TOKEN: ${{ secrets.BALLERINA_BOT_TOKEN }}
               run: |
-                  curl -fsSL https://github.com/github/hub/raw/master/script/get | bash -s 2.14.1
-                  bin/hub pull-request --base main -m "[Automated] Sync main after "$VERSION" release"
+                gh release create v$VERSION --title "module-ballerina-random-v$VERSION"
+                gh pr create --title "[Automated] Sync master after $VERSION release" --body "Sync master after $VERSION release"
```
Comment (ThisaruGuruge) [inline .github/workflows/publish-release.yml:53]:
```suggestion
                gh pr create --base main --title "[Automated] Sync master after $VERSION release" --body "Sync master after $VERSION release"
```

**Comment by MadhukaHarith92** [comment]
> Gradle Wrapper jar is not pushed. Please check

Updated in f14d16a

File: .github/workflows/publish-release.yml [hunk 4]
```diff
@@ -45,19 +45,9 @@ jobs:
...
-                  }'
-            - name: Post release PR
+            - name: GitHub Release and Release Sync PR
               env:
                   GITHUB_TOKEN: ${{ secrets.BALLERINA_BOT_TOKEN }}
               run: |
-                  curl -fsSL https://github.com/github/hub/raw/master/script/get | bash -s 2.14.1
-                  bin/hub pull-request --base main -m "[Automated] Sync main after "$VERSION" release"
+                gh release create v$VERSION --title "module-ballerina-random-v$VERSION"
+                gh pr create --title "[Automated] Sync main after $VERSION release" --body "Sync main after $VERSION release"
```
Comment (ThisaruGuruge) [inline .github/workflows/publish-release.yml:53]:
```suggestion
                gh pr create --base main --title "[Automated] Sync main after $VERSION release" --body "Sync main after $VERSION release"
```

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**